spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/RegionRroposal.scala

```diff
@@ -0,0 +1,342 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn
+
+import java.util
+import breeze.linalg.{dim, min}
+import com.intel.analytics.bigdl.Module
+import com.intel.analytics.bigdl.nn.abstractnn.AbstractModule
+import com.intel.analytics.bigdl.tensor.{Storage, Tensor}
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.transform.vision.image.util.BboxUtil
+import com.intel.analytics.bigdl.utils.{LayerException, T, Table}
+import scala.collection.mutable.ArrayBuffer
+
+/**
+ * Layer for RPN computation. Takes feature maps from the backbone and
+ * outputs RPN proposals and losses.
+ * @param inChannels
+ * @param anchorSizes
+ * @param aspectRatios
+ * @param anchorStride
+ * @param preNmsTopNTest
+ * @param postNmsTopNTest
+ * @param nmsThread
+ * @param minSize
+ */
+class RegionRroposal(
+   val inChannels: Int,
+   val anchorSizes: Array[Float],
+   val aspectRatios: Array[Float],
+   val anchorStride: Array[Float],
+   val preNmsTopNTest: Int = 1000,
+   val postNmsTopNTest: Int = 1000,
+   val preNmsTopNTrain: Int = 2000,
+   val postNmsTopNTrain: Int = 2000,
+   val nmsThread: Float = 0.7f,
+   val minSize: Int = 0)(implicit ev: TensorNumeric[Float])
+   extends AbstractModule[Table, Tensor[Float], Float] {
```

Comment:
why harcoded to Float ?

Reply:
for performance and code style, it's better to use float.